test/e2e/kubectl/kubectl.go

```diff
@@ -2044,8 +2080,15 @@ func checkOutputReturnError(output string, required [][]string) error {
 			return fmt.Errorf("failed to find %s in %s", requirement[0], output)
 		}
 		for _, item := range requirement[1:] {
-			if !strings.Contains(outputLines[currentLine], item) {
-				return fmt.Errorf("failed to find %s in %s", item, outputLines[currentLine])
+			if _, err := strconv.Atoi(item); err == nil {
+				// When item has only numerical value, check if it is contained at the suffix of the line for strict check.
```

Comment:
Sorry, I could not understand this code.
I guess the additional `{"Successful Job History Limit:", "3"},` of line 1150 can detect the issue of https://github.com/kubernetes/kubernetes/pull/77347 even if this check is not added, right?

Reply:
Thanks for the comment.
Original code checks if expect value is contained. For example, when expect value is "3" and actual value is "32", original code returns "true". So, this commit checks a value at suffix strictly. Of course, if expect value is "3" and actual "23", changed code returns "true". But, change to check more strictly affects existing tests.

Reply:
Thanks for your explanation, I see the point.
How about
```
if expected_value, err := strconv.Atoi(item); err == nil {
    actual_value, err := strconv.Atoi(outputLines[currentLine])
    if err != nil || expected_value != actual_value {
        return fmt.Errorf("failed to find %s in %s", item, outputLines[currentLine])
    }
```
for more strict check?

Reply:
Thanks for the good proposal.
I apologize that the content I told you was not enough.
The function checks entire row not just value which `kubectl describe` outputs.
For example, it checks if "Successful Job History Limit:  3"(actual value) contains "Successful Job History Limit:". After that, it checks "Successful Job History Limit:  3" contains "3".

I'm considering the following change.
```
if _, err := strconv.Atoi(item); err == nil {
        if strings.TrimSpace(strings.Replace(outputLines[currentLine], requirement[0], "", 1)) != item {
                return fmt.Errorf("failed to find %s in %s", item, outputLines[currentLine])
        }
```
Applying the above example, outputLines[currentLine] = "Successful Job History Limit:  3", requirement[0] = "Successful Job History Limit:", so I think value can be checked.

Reply:
@oomichi Could you review the latest commit?

Reply:
@k-toyoda-pi I still feel the additional check code seems a little tricky and a little hard to maintain it in long-term.
In addition, checkOutputReturnError() is used for conformance tests also which should be stable always.
How about separating the additional check of checkOutputReturnError() from this PR and proposing another PR?
The additional test `should check if kubectl describe prints relevant information for cronjob` seems simply good for me.

Reply:
Thanks the comment. I agree with you. I've reverted `checkOutputReturnError`.